src/aof.c

```diff
@@ -1551,6 +1551,9 @@ int loadAppendOnlyFiles(aofManifest *am) {
     }
 
     if (am->base_aof_info == NULL && listLength(am->incr_aof_list) == 0) {
+        /* If we start with an empty dataset and ‘aof_use_rdb_preamble’ is set to yes, we 
+         * will force create a BASE (rdb format) file. */
+        if (server.aof_use_rdb_preamble) server.aof_rewrite_scheduled = 1;
```

Comment:
@guybe7 @MeirShpilraien @yossigo can you think of any disadvantage, or advantage in triggering a real rewrite (fork) this way, rather than calling a foreground rdbSave?

Reply:
The code has been changed to the foreground rdbSave, I think I implemented it with a small cost.